llm/llama.go

```diff
@@ -184,16 +185,62 @@ type llama struct {
 	Running
 }
 
-var errNoGPU = errors.New("nvidia-smi command failed")
+var errNoNVIDIAGPU = errors.New("nvidia-smi command failed")
+var errNoAMDGPU = errors.New("rocm-smi command failed")
+var errNoSupportedGPU = errors.New("no supported GPU found")
 
-// CheckVRAM returns the available VRAM in MiB on Linux machines with NVIDIA GPUs
+// CheckVRAM returns the available VRAM in MiB on Linux machines with nVidia or AMD GPUs.
 func CheckVRAM() (int64, error) {
+	for _, f := range []func() (int64, error){CheckVRAMNVIDIA, CheckVRAMAMD} {
+		if vram, err := f(); err == nil && vram > 0 {
+			return vram, nil
+		}
+	}
+	return 0, errNoSupportedGPU
+}
+
+// CheckVRAMAMD returns the available VRAM in MiB on Linux machines with AMD GPUs
+func CheckVRAMAMD() (int64, error) {
+	rocmHome := os.Getenv("ROCM_HOME")
+	if rocmHome == "" {
+		rocmHome = "/opt/rocm"
+	}
+	cmd := exec.Command(filepath.Join(rocmHome, "bin/rocm-smi"), "--showmeminfo", "VRAM", "--csv")
+	var stdout bytes.Buffer
+	cmd.Stdout = &stdout
+	err := cmd.Run()
+	if err != nil {
+		return 0, errNoAMDGPU
```

Comment:
```suggestion
                log.Printf("rocm-smi failed: %s", err)
		return 0, errNoAMDGPU
```

Just in-case we need to debug this

Reply:
Easier to just put the printf in an else in CheckVRAM I think, and let it bubble up there. It'll get two birds with one stone. I'll add that and you can see what I mean.

Reply:
Added, please take a look and let me know what you think.

Reply:
I think this is addressed, but feel free to reopen. `CheckVRAM` will now notify the user if any of the subcommands fail. This is a little noiser for the user, but hopefully the message is clear that the check is skipped and there is no actual error.